mlflow/utils/databricks_utils.py

```diff
@@ -168,7 +168,7 @@ def is_in_databricks_job():
 
 
 def is_in_databricks_model_serving_environment():
-    return "DATABRICKS_MODEL_SERVING_ENV" in os.environ
+    return "IS_IN_DATABRICKS_MODEL_SERVING_ENV" in os.environ
```

Comment:
We need to rename the env var used to determine whether MLflow is running Databricks Model Serving because clients running MLflow 2.11.1 or 2.11.2 will break if an env var with the previous env var name is set.